src/open_inwoner/components/templatetags/kvk_tags.py

```diff
@@ -0,0 +1,180 @@
+import json
+
+from django import template
+from django.utils.html import escape
+from django.utils.safestring import mark_safe
+
+register = template.Library()
+
+
+@register.simple_tag
+def react_branch_data(branches, selected_id=None):
+    """
+    Convert Django branch data to React-compatible JSON format
+    This matches exactly what SelectComboBoxModule expects
+
+    This function creates JSON data that gets embedded in HTML and read by React.
+    The 'selected_id' corresponds to the branch_number field that Django expects
+    when the form is submitted via POST request.
+    """
+    items = []
+    for branch in branches:
+        # Handle the "entire company" case (empty vestigingsnummer)
+        # This becomes the 'id' that React stores and sends back as 'branch_number'
+        branch_id = branch.get("vestigingsnummer", "") or "entire-company"
```

Comment:
```suggestion
        branch_id = branch.get("vestigingsnummer", "rechtspersoon")
```

You can add the default as second argument to `get`. I would use the Dutch "rechtspersoon" because it is clear ("entire company" occurs elsewhere but really shouldn't have been used to begin with; it's not very clear)

Reply:
But if I do this than I will lose the Rechtspersoon in case of empty string, so if vestigingsnummer is missing, use 'rechtspersoon'". BUT if vestigingsnummer exists and is an empty string (""), it would use that empty string instead of "rechtspersoon"